packages/react-instantsearch-core/src/core/utils.ts

```diff
@@ -33,13 +33,19 @@ export const defer = f => {
   resolved.then(f);
 };
 
-export const removeEmptyKey = obj => {
+const isPlainObject = (value: any): value is object =>
```

Comment:
I'm not sure if this type assertion can / needs to be any better

Reply:
I think this is the correct way to do this, TypeScript seems to do the same thing in [the documentation](https://www.typescriptlang.org/docs/handbook/advanced-types.html#typeof-type-guards)

Reply:
I think this is were you're supposed to use `unknown` ([see resource I sent yesterday](https://mariusschulz.com/blog/typescript-3-0-the-unknown-type#narrowing-the-unknown-type)):

```suggestion
const isPlainObject = (value: unknown): value is object =>
```